cmd/tendermint/commands/util.go

```diff
@@ -0,0 +1,160 @@
+package commands
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+)
+
+// indexTaggedStructFields walks through a struct, looking up
+// those fields tagged by the requested tagName e.g:
+//    Foo int `json:"foo"`
+//    Bar string `mapstructure:"bar"`
+// recursively flattening anonymous struct fields by calling indexTaggedStructFields
+// on them.
+func indexTaggedStructFields(v interface{}, tagName string) map[string]interface{} {
+	val := reflect.ValueOf(v)
+	fieldsMap := make(map[string]interface{})
+	if val.Kind() == reflect.Ptr {
+		val = reflect.Indirect(val)
+	}
+
+	typ := val.Type()
+	n := val.NumField()
+
+	for i := 0; i < n; i++ {
+		fVal := val.Field(i)
+		field := typ.Field(i)
+		tag, _, squash := extractTag(tagName, field)
+		if tag != "" {
+			fieldsMap[tag] = fVal.Interface()
+		} else if squash {
+			innerMap := indexTaggedStructFields(fVal.Interface(), tagName)
+			// Since this is an anonymous struct,
+			// we should squash as requested it's content inside
+			for k, v := range innerMap {
+				if _, alreadyIn := fieldsMap[k]; !alreadyIn {
+					fieldsMap[k] = v
+				}
+			}
+		}
+	}
+
+	return fieldsMap
+}
+
+func extractTag(tagName string, f reflect.StructField) (theTagName string, omitEmpty, squash bool) {
+	tag := f.Tag.Get(tagName)
+	splits := strings.Split(tag, ",")
+	if len(splits) == 0 {
+		return "", false, false
+	}
+	if len(splits) > 1 {
+		switch splits[1] {
+		case "omitempty":
```

Comment:
what if both of them are present? `omitempty,squash`

Reply:
So far I haven't seen an inclusion of both in "mapstructure" as "squash" is the equivalent "omitempty" and the idea is that they are mutually exclusive.

Reply:
We don't use both anywhere but I don't see why they would be mutually exclusive? 

`squash` brings a substruct up to the current level in json serialization. `omitempty` leaves it out all together. I could imagine using both.